R/matching.R

```diff
@@ -154,35 +156,85 @@ plot_weights <- function(wt, main_title = "Unscaled Individual Weights") {
 #' before and after adjustment.
 #'
 #' @param optimized object returned after calculating weights using \code{\link{estimate_weights}}
-#' @param match_cov covariates that should be checked to see if the IPD weighted average matches the aggregate data
-#' average. This could be same set of variables that were used to match or it can include variables that were not
-#' included to match (i.e. stratification variables)
+#' @param processed_agd a data frame, object returned after using \code{\link{process_agd}} or aggregated data following the same naming convention
 #' @param digits number of digits for rounding summary table
 #'
+#' @import DescTools
+#'
 #' @return data.frame of weighted and unweighted covariate averages of the IPD
 #' @export
 
-
-
-check_weights <- function(optimized, match_cov, digits = 2) {
-  if (missing(match_cov)) stop("match_cov is missing. Covariates to check must be defined.")
+check_weights <- function(optimized, processed_agd, mean_digits = 2, prop_digits = 2, sd_digits = 3) {
   ipd_with_weights <- optimized$data
-
-  arm_names <- c("Unweighted IPD", "Weighted IPD")
-  ess <- c(nrow(ipd_with_weights), optimized$ess)
-
-  unweighted_cov <- sapply(ipd_with_weights[, match_cov, drop = FALSE], mean)
-
-  weighted_cov <- sapply(
-    ipd_with_weights[, match_cov, drop = FALSE],
-    weighted.mean,
-    w = ipd_with_weights$weights
+  match_cov <- optimized$centered_colnames
+
+  # if algorithm is correct, all centered columns should have a weighted summation to a very small number around zero
+  num_check <- ipd_with_weights$weights %*% as.matrix(ipd_with_weights[, match_cov, drop = FALSE])
+  num_check <- round(num_check, 4)
+
+  # for reporting
+  outdata <- data.frame(
+    covariate = gsub("_CENTERED$", "", match_cov),
+    match_stat = NA,
+    external_trial = NA,
+    internal_trial = NA,
+    internal_trial_after_weighted = NA,
+    sum_centered_IPD_with_weights = as.vector(num_check)
```

Comment:
```suggestion
  outdata <- data.frame(
    covariate = gsub("_CENTERED$", "", match_cov),
    match_stat = NA,
    internal_trial = NA,
    internal_trial_after_weighted = NA,
     external_trial = NA,
    sum_centered_IPD_with_weights = as.vector(num_check)
```
I would suggest that the 'external_trial' column goes after the weighted column so that we can more easily see if the results of these two columns are the same in the final outdata output. 

Also, do we need the  sum_centered_IPD_with_weights column? Wouldn't be sufficient to check only whether the re-weighted baseline characteristics for the intervention-treated patients match those aggregate characteristics from the comparator trial?

Reply:
Yea I also think external should be next to internal adjusted for easier comparison. I can put that in when I do a pull. 
I thought sum_centered_IPD_with_weights term was cool. I didn't think about this before. I think it is another measure to see if optimization has worked.